src/client-agent/notify.c

```diff
@@ -42,30 +42,19 @@ char *getsharedfiles()
 #ifndef WIN32
 char *get_agent_ip()
 {
-    static char agent_ip[IPSIZE + 1] = { '\0' };
+    char agent_ip[IPSIZE + 1] = { '\0' };
```

Comment:
```suggestion
    char agent_ip[IPSIZE] = { '\0' };
```

Reply:
You should always allocate an extra byte if you use OS_RecvUnix, since in line 479 it puts a \0 at the end of the message based on the size. If we don't do it, the stack will be corrupted

https://github.com/wazuh/wazuh/blob/edeb91437fc43810781503ab0a7b492392766559/src/os_net/os_net.c#L479

and if we do an IPSIZE-1 as an argument to avoid stack corruption we would truncate the message in the case that it is an ipv4 as ipv6 (ex: "abcd:abcd:abcd:abcd:abcd:abcd:192.168.100.100") because of the subtraction(recvfrom) on the line:
https://github.com/wazuh/wazuh/blob/edeb91437fc43810781503ab0a7b492392766559/src/os_net/os_net.c#L481

in the case of truncating the ip it would be "abcd:abcd:abcd:abcd:abcd:abcd:192.168.100.10"